clients/src/main/java/org/apache/kafka/clients/producer/internals/TransactionManager.java

```diff
@@ -1073,6 +1074,11 @@ private void transitionTo(State target, RuntimeException error) {
         } else if (target == State.FATAL_ERROR || target == State.ABORTABLE_ERROR) {
             if (error == null)
                 throw new IllegalArgumentException("Cannot transition to " + target + " with a null exception");
+
+            if (error instanceof RetriableException) {
```

Comment:
Can we leave a comment that RetriableExceptions from the Sender thread should be translated to abortable?